### PR TITLE
Navigation fragment decode and encodings

### PR DESCRIPTION
**Pull request by annevk**
See https://github.com/whatwg/html/pull/3111 for context.

### CODE REVIEW

**Comment by w3c-bots** [comment]
# Build PENDING

Started: None
Finished: None





View more information about this build on:

- [WPT PR Status](http://pulls.web-platform-tests.org/build/21919)
- [TravisCI](https://travis-ci.org/w3c/web-platform-tests/builds/318050123)

**Comment by w3c-bots** [comment]
# Build PENDING

Started: None
Finished: None





View more information about this build on:

- [WPT PR Status](http://pulls.web-platform-tests.org/build/22718)
- [TravisCI](https://travis-ci.org/w3c/web-platform-tests/builds/328059925)

File: html/browsers/browsing-the-web/scroll-to-fragid/fragment-and-encoding-2.html [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+<div style=height:10000px></div>
+<div id=&#xFFFD;></div>
+<script>
+test(() => {
+  assert_equals(location.hash, "", "Page must be loaded with no hash");
+
+  location.hash = "%C2";
+  assert_equals(location.hash, "#%C2");
+  assert_greater_than(self.scrollY, 1000, "#%C2"); // UTF-8 decoded this'll match U+FFFD
+
```
Comment (tkent-google) [inline html/browsers/browsing-the-web/scroll-to-fragid/fragment-and-encoding-2.html]: IMO, this test should be split into two or three.

test(() => {
 .... %C2
 ... %EF%BB%BF%C2
}, 'Invalid byte for UTF-8 should produce U+FFFD');

test(() => {
  .. .%EF%BF%BD
}, 'Percent-deocde result should be decoded as UTF-8 even if document encoding is windows-1252');

}, '

File: html/browsers/browsing-the-web/scroll-to-fragid/fragment-and-encoding-2.html [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  location.hash = "top";
+  assert_equals(self.scrollY, 0, "#top");
+
+  location.hash = "%EF%BB%BF%C2";
+  assert_equals(location.hash, "#%EF%BB%BF%C2");
+  assert_equals(self.scrollY, 0, "#%EF%BB%BF%C2");
```
Comment (tkent-google) [inline html/browsers/browsing-the-web/scroll-to-fragid/fragment-and-encoding-2.html]: Probably we should have an element with U+FEFF U+00C2 to show the Chrome's bug?

File: html/browsers/browsing-the-web/scroll-to-fragid/fragment-and-encoding.html [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  assert_greater_than(self.scrollY, 1000, "#%C3%BF");
+
+  location.hash = "top";
+  assert_equals(self.scrollY, 0, "#top");
+
+  location.hash = "%EF%BB%BF";
+  assert_greater_than(self.scrollY, 1000, "#%EF%BB%BF");
+
+  location.hash = "%FF";
+  assert_equals(self.scrollY, 0, "#%FF");
```
Comment (tkent-google) [inline html/browsers/browsing-the-web/scroll-to-fragid/fragment-and-encoding.html]: Is this correct?  self.scrollY before the hash assignment should be 1000+, and there is no element with id=U+FFFD or "%FF".  This should not change scrollY, should it?

**Comment by annevk** [comment]
Removing "do not merge yet" as we made a decision in the HTML Standard.

@tkent-google I addressed your feedback I think. Mind taking another look?

**Review by tkent-google — APPROVED** [review]

**Pull request merged**